ft_virtualchannel.m

```diff
@@ -0,0 +1,382 @@
+    montage.labelnew = cell(size(montage.tra,1), 1);
+    cnt = 0;
+    for k = 1:numel(sel)
+      for kk = 1:size(source.filter{sel(k)},1)
+        cnt = cnt+1;
+        montage.labelnew{cnt} = sprintf('%s_dipole%03d_orientation%03d', cfg.parcel{i}, k, kk);
+      end
+    end
+    bname = matlab.lang.makeValidName(cfg.parcel{i}); % FIXME not sure how universal this is.
+    
+  end
+  
+  % apply the montage to the numeric data, but make an exception for the
+  % svd method: this one is much more efficient, in that its implementattion
+  % operates on the covariance matrix directly, bypasses ft_componentanalysis, 
+  % and does not require the intermediate (possibly memory greedy, and slow)
+  % data representation.
+  if ~isequal(cfg.method, 'svd')
+    tmpdata{1,i}  = ft_apply_montage(data, montage, 'feedback', 'none');
+    
+    [i1, i2] = match_str(data.label, montage.labelold);
+    unmixing{1,i}(:,i1) = montage.tra(:,i2);
+    topolabel{1,i}      = data.label(i1);
+    
+    % apply the montage to the sensor description
+    sensfields = {'grad' 'elec' 'opto'};
+    for k = 1:numel(sensfields)
+      if isfield(tmpdata{i}, sensfields{k})
+        ft_info(sprintf('applying the montage to the %s structure\n', sensfields{k}));
+        tmpdata{i}.(sensfields{k}) = ft_apply_montage(tmpdata{i}.grad, montage, 'feedback', 'none', 'keepunused', 'no', 'balancename', bname);
+      end
+    end
+    
+    compmethods = {'pca' 'runica' 'fastica' 'dss'};
+    switch cfg.method
+      case compmethods
+        
+        tmpcfg     = keepfields(cfg, {'method' cfg.method 'numcomponent'});
+        if isequal(tmpcfg.numcomponent, 'all')
+          tmpcfg = rmfield(tmpcfg, 'numcomponent');
+        end
+        
+        tmpdata{i}  = ft_componentanalysis(tmpcfg, tmpdata{i});
+        unmixing{i} = tmpdata{i}.unmixing * unmixing{i};
+        
+        for k = 1:numel(tmpdata{i}.label)
+          if usepos
+            str = sprintf('virtualchannel%03d', i);
+          else
+            str = cfg.channel{i};
+          end
+        end
+        tmpdata{i}.label{k} = sprintf('%s_%s%03d', str, cfg.method, k);
+      
+      case 'none'
+        % do nothing
+      
+    otherwise
+      ft_error('currently not yet supported');
+      % the idea would to support a custom function here, with a
+      % function(cfg.(cfg.method), tmpdata{i}) API
+      
+    end % reduction of components
+  
+  else
+    
+    [i1, i2] = match_str(data.label, montage.labelold);
+    
+    % ensure that the used channels run from 1:numel(data.label), to be
+    % sure that the to-be-computed covariance matches the labels, not sure
+    % whether this is needed, this could probably also be achieved by doing
+    % a check on the input data's labels, related to the source.filter
+    % labels
+    assert(isequal(i1(:)', 1:numel(data.label)));
+    
+    unmixing{1,i}(:,i1) = montage.tra(:,i2);
+    topolabel{1,i}      = data.label(i1);
+    
+    if ~exist('tlck', 'var')
```

Comment:
```suggestion
    if ~exist('tlck', 'cov')
```
It's possible that `tlck` does have a `var` but no `cov`.

Reply:
indeed, but the variable needs to exist, so a separate check needs to be done using isfield.

Reply:
Nvm, I was not reading properly here.